File: scripts/load-50-spins.mjs
#!/usr/bin/env node
/**
 * Carica gli ultimi numeri (roulette 0-36) nell'API.
 * Ordine: dal più vecchio al più recente.
 * Uso: node scripts/load-50-spins.mjs [URL_API]
 */

const API_BASE = (process.env.API_URL || process.argv[2] || 'http://localhost:8000').replace(/\/$/, '');

// Last Results: 50 numeri, 5 righe x 10. Ordine dal più vecchio al più recente (9 = ultimo).
const NUMBERS = [
  17, 31, 25, 17, 32, 32, 24, 30, 23, 32,
  2, 28, 34, 16, 13, 0, 19, 30, 29, 2,
  15, 30, 18, 11, 2, 26, 1, 9, 13, 7,
  32, 17, 33, 31, 1, 23, 12, 28, 21, 19,
  14, 0, 28, 23, 11, 32, 8, 28, 0, 9,
];

async function main() {
  console.log(`Caricamento di ${NUMBERS.length} uscite su ${API_BASE}...`);
  let ok = 0;
  let err = 0;
  for (let i = 0; i < NUMBERS.length; i++) {
    const n = NUMBERS[i];
    try {
      const res = await fetch(`${API_BASE}/spins`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ number: n }),
      });
      if (!res.ok) throw new Error(await res.text());
      ok++;
      process.stdout.write(`  ${i + 1}/${NUMBERS.length} → ${n} ${res.ok ? '✓' : ''}\n`);
    } catch (e) {
      err++;
      console.error(`  ${i + 1}/${NUMBERS.length} → ${n} ERRORE:`, e.message);
    }
    // piccola pausa per non stressare il server
    await new Promise((r) => setTimeout(r, 80));
  }
  console.log(`\nFatto: ${ok} ok, ${err} errori.`);
  if (err > 0) process.exit(1);
}

main();

File: scripts/load-50-spins.py
#!/usr/bin/env python3
"""
Carica 110 numeri casuali (roulette 0-36) nell'API per test completi.
Seed fisso 42 = stessa sequenza ogni volta.
Uso: python scripts/load-50-spins.py [URL_API]
"""
import sys
import time
import urllib.request
import json

API_BASE = (sys.argv[1] if len(sys.argv) > 1 else "http://localhost:8000").rstrip("/")

# 110 numeri casuali (seed 42), ordine dal più vecchio al più recente
NUMBERS = [
    7, 1, 17, 15, 14, 8, 6, 34, 5, 27, 2, 1, 5, 13, 14, 32, 1, 35, 12, 34, 26, 14, 28, 17, 0, 10, 27, 21, 17, 9, 13, 21, 6, 5, 24, 6, 22, 22, 16, 2, 29, 34, 7, 24, 5, 35, 18, 23, 36, 12, 4, 2, 14, 18, 5, 14, 6, 24, 17, 29, 23, 10, 23, 22, 13, 17, 4, 10, 34, 15, 10, 29, 24, 17, 35, 14, 20, 3, 14, 2, 20, 25, 17, 4, 13, 36, 20, 13, 31, 25, 29, 9, 16, 8, 15, 35, 34, 16, 27, 25, 23, 14, 8, 32, 31, 5, 3, 7, 9, 10,
]


def main():
    print(f"Caricamento di {len(NUMBERS)} uscite su {API_BASE}...")
    ok = err = 0
    for i, n in enumerate(NUMBERS):
        try:
            req = urllib.request.Request(
                f"{API_BASE}/spins",
                data=json.dumps({"number": n}).encode(),
                headers={"Content-Type": "application/json"},
                method="POST",
            )
            with urllib.request.urlopen(req) as res:
                if res.status == 200:
                    ok += 1
                    print(f"  {i + 1}/{len(NUMBERS)} → {n} ✓")
                else:
                    raise Exception(f"HTTP {res.status}")
        except Exception as e:
            err += 1
            print(f"  {i + 1}/{len(NUMBERS)} → {n} ERRORE: {e}")
        time.sleep(0.08)
    print(f"\nFatto: {ok} ok, {err} errori.")
    sys.exit(1 if err else 0)


if __name__ == "__main__":
    main()
